avogadro/core/molecule.cpp

```diff
@@ -825,27 +825,27 @@ void Molecule::perceiveBondsSimple(const double tolerance, const double min)
     )
   );
   std::vector<std::array<size_t, 3>> atoms_bin(atomCount());
-  for (Index i = 1; i < atomCount(); i++) {
+  for (Index i = 0; i < atomCount(); i++) {
     Vector3 ipos = m_positions3d[i];
     std::array<size_t, 3> bin_index;
     for (size_t c = 0; c < 3; c++) {
       bin_index[c] = std::floor((ipos(c) - min_pos(c)) / bin_size);
     }
     bins.at(bin_index[0]).at(bin_index[1]).at(bin_index[2]).push_back(i);
-    atoms_bin.push_back(bin_index);
+    atoms_bin[i] = bin_index;
   }
 
   // check for bonds
   // O(n) average-case, O(n^2) worst-case
   // note that the "worst case" here would need to be an invalid molecule
   for (Index i = 0; i < atomCount(); i++) {
     Vector3 ipos = m_positions3d[i];
-    std::array<size_t, 3> bin_index = atoms_bin[i];
-    for (Index xi = std::max(size_t(0), bin_index[0] - 1);
```

Comment:
The `const` is good, but why did you change the rest of this? It should stay `size_t` rather than a `signed` - I'm getting compiler warnings from `clang`

Reply:
The problem is, `size_t` is an unsigned type, so `std::max(size_t(0), bin_index[0] - 1)` will produce weird results if `bin_index[0]` equals 0. Another option would be  `std::max(size_t(1), bin_index[0]) - 1`, which avoids the cast. I'll write that instead.